src/preact/base-element.js

```diff
@@ -615,21 +757,30 @@ export class PreactBaseElement extends BaseElement {
         this.container_ = container;
 
         // Connect shadow root to the element's context.
+        devAssert(shadowRoot);
         setParent(shadowRoot, this.element);
         // In Shadow DOM, only the children distributed in
         // slots are displayed. All other children are undisplayed. We need
         // to create a simple mechanism that would automatically compute
         // `CanRender = false` on undistributed children.
         addGroup(this.element, UNSLOTTED_GROUP, MATCH_ANY, /* weight */ -1);
-        // eslint-disable-next-line local/restrict-this-access
-        setGroupProp(this.element, UNSLOTTED_GROUP, CanRender, this, false);
+
+        setGroupProp(
+          this.element,
+          UNSLOTTED_GROUP,
+          CanRender,
+          // TODO: is `this` correct as the setter arg for setGroupProp
+          // eslint-disable-next-line local/restrict-this-access
+          /** @type {*} */ (this),
+          false
+        );
       } else if (lightDomTag) {
-        this.container_ = this.element;
+        const container = (this.container_ = this.element);
```

Comment:
This assignment-assignment is harder to read imo and only saves one `this.` access, doesn't feel worthwhile to me

Reply:
Ok, will split into two lines.